CalibFormats/HcalObjects/src/HcalText2DetIdConverter.cc

```diff
@@ -229,10 +239,14 @@ bool HcalText2DetIdConverter::init (const std::string& fFlavor, const std::strin
     int channel = calibChannel (field3);
     mId = HcalCalibDetId (sd, ieta,iphi,channel);
   }
-  else if (flavorName=="HOX") {
+  else if (flavorName=="HOX" || flavorName=="HBX" || flavorName=="HEX") {
     int ieta=getField(1);
     int iphi=getField(2);
-    mId = HcalCalibDetId (ieta,iphi);
+    mId = (flavorName=="HOX")?
```

Comment:
This construction is not optimal: why the same string is compared twice?

Reply:
@civanch @abdoulline I agree this logic might made a bit simpler by splitting the three new cases into separate actions...

Reply:
@fabiocos  Fabio, I hope it doesn't prevent the PR from being merged as is? 
HCAL DPG is waiting for it to proceed with updating emap (extended for HBX/HEX channels).

Reply:
@abdoulline I will integrate for this IB, not the cleanest logic to my mind, but if it is working correctly fine, it could be cleaned in some further pass

Reply:
@fabiocos  it behaves/works correctly, the way it coded can be made more explicit/translarent next time.